src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -1396,6 +1402,29 @@ internal static bool OnServerCertificateCustomValidationCallback(object sender,
             return false;
         }
 
+        private static void ThrowNameMismatchException(object sender, X509Certificate cert)
+        {
+            var hostname = ((HttpRequestMessage)sender).RequestUri.DnsSafeHost;
```

Comment:
It is not always HttpRequestMessage. We are using this callback also in Changes API with WebSockets. Then the sender is SslStream